app/models/id_finder.rb

```diff
@@ -0,0 +1,11 @@
+class IdFinder
+  def self.go(session, search_text)
+    (session[:search_memo][search_text] rescue nil) || query_for_id(search_text)
+  end
+
+  def self.query_for_id(search_text)
+    @client = GooglePlaces::Client.new(ENV['GOOGLE_KEY'])
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.